client/collator/src/lib.rs

```diff
@@ -241,6 +241,40 @@ where
 				}
 			};
 
+			let code_hash = sp_io::storage::get(well_known_keys::CODE_HASH);
+			let code_hash = if let Some(code_hash) = code_hash {
+				if code_hash.len() != Hash::len_bytes() {
+					tracing::error!(
+						target: LOG_TARGET,
+						"Invalid code hash in storage: len is {}, expected {}",
+						code_hash.len(),
+						Hash::len_bytes(),
+					);
+					return None;
+				}
+
+				Hash::from_slice(code_hash.as_slice())
+			} else {
+				tracing::info!(
+					target: LOG_TARGET,
+					"Failed to get validation code hash from storage, hash validation code \
+					directly."
+				);
+
+				let code = sp_io::storage::get(sp_core::storage::well_known_keys::CODE);
+				let code = match code {
+					Some(code) => ValidationCode(code),
+					None => {
+						tracing::error!(
+							target: LOG_TARGET,
+							"Failed to get validation code."
+						);
+						return None;
+					}
+				};
+				code.hash()
```

Comment:
this branch when code is not found could be removed once all current runtimes have upgraded and put their code hash in storage.